contents/tutorials/validating-what-you-ship.md

```diff
@@ -0,0 +1,224 @@
+
+Without pre-defined success criteria, you end up in one of two traps:
+
+1. **The moving goalpost**: You look at the data post-launch and adjust your definition of success to match what happened. "Well, we didn't hit 20% adoption, but 8% is pretty good for a v1."
+
+2. **The infinite iteration loop**: You keep tweaking because you're not sure if it's working. Six months later, you're still on "v1.4" of something that should have been killed or promoted months ago.
+
+The fix is simple but requires discipline: write down what success looks like before you ship, then hold yourself to it.
+
+## Validation by example
+
+Let's say you're launching a new onboarding flow for your SaaS product.
+
+**Before launch**, you define success:
+
+- Primary metric: Activation rate (users who complete onboarding AND perform a key action within 7 days)
+- Target: Increase from 35% to 45%
+- Timeline: Measured 4 weeks post-launch
+- Guardrail: Support ticket volume doesn't increase by more than 10%
+
+**At launch**, you ship to a subset of users first. Create a feature flag called `new-onboarding-flow` and roll it out to 20% of users. This gives you a natural control group for comparison.
+
+**Post-launch**, the data tells a clear story. Activation rate went from 35% to 41%. Not the 45% you hoped for, but a measurable improvement. You also notice step 3 of onboarding has a 40% drop-off. That's your iteration target.
+
+Without this setup, you'd be guessing. With it, you know exactly what happened and what to do next.
+
+## Setting up validation in PostHog
+
+### Step 1: Write your hypothesis
+
+Before writing any code, document:
+
+- **What are you changing?** (Be specific: "New onboarding flow with interactive tutorial")
+- **What do you expect to happen?** ("Users will activate at a higher rate because they understand the product faster")
+- **How will you measure it?** ("Compare activation rate between users who see new flow vs. old flow")
+- **What's your success threshold?** ("10 percentage point improvement in activation rate")
+- **What could go wrong?** ("Users might find the tutorial annoying and abandon earlier")
+
+> **Planning prompt**: What could go wrong? Write this down alongside your success condition.
+
+### Step 2: Define your metrics
+
+**Leading indicators** tell you if users are engaging with your change:
+
+| Metric | Example event | What it tells you |
+|--------|---------------|-------------------|
+| Discovery | `onboarding_started` | Are users entering the flow? |
+| Progress | `onboarding_step_completed` | Are users moving through it? |
+| Completion | `onboarding_completed` | Are users finishing? |
+
+> **Tip:** When capturing events, include a `step_name` property so you can later analyze where users get stuck.
+
+[Complete guide to event tracking](/tutorials/event-tracking-guide)
```

Comment:
```suggestion
If you're need help capturing the necessary events, try the [complete guide to event tracking](/tutorials/event-tracking-guide)
```

The link was just getting swallowed visually, I think this will help, if not, try an OS button?